unit_tests/bigquery/test_table.py

```diff
@@ -1449,6 +1449,12 @@ class TextModeFile(object):
         with self.assertRaises(ValueError):
             table.upload_from_file(file_obj, 'CSV', size=1234)
 
+    def test_upload_from_file_binary_mode_success(self):
+        try:
+            self._upload_from_file_helper(input_file_mode='r+b')
+        except ValueError:
+            self.fail("Unexpected failure when uploading binary mode file")
```

Comment:
Right, I don't like it much either (see previous comment about it making coverage tests fail, too).

The test is only to ensure that `r+b` doesn't raise an exception. I can't think of an elegant way of doing this. Skipping the `try`/`except` would mean that the test would Error rather than Fail, which isn't correct. Additionally, checking that the file is created (for example), is only indirectly exercising the bug.

Suggestions appreciated...